hammurabi/rules/text.py

```diff
@@ -155,7 +152,9 @@ def task(self) -> Path:
 
         if not file_was_empty and no_criteria_match:
             target_match = self.__get_target_match(lines)
-            target_match_index = lines.index(target_match)
+
+            # Get the index of the element from the right
```

Comment:
Before this change, the LineExists raised an exception if multiple matches found. This would not be a bad idea, but I did not count with the case: What if we want to append something at the end of the file?

What happens now (and before) is that we read the file, and split it by lines, which is good for us, because we can iterate every line, check for matches and so on. The only (and huge) issue was that we were not able to add something to the beginning or end of the file.

This change will ensure that we will insert after the last match if we found multiple matches for the target. This way we can insert anything to the end of the file (which, again, is a list of splitted lines) 😊

Reply:
hm, right